numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -1583,7 +1583,7 @@ _array_fromobject(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kws)
 
     if (PyTuple_GET_SIZE(args) > 2) {
         PyErr_SetString(PyExc_ValueError,
-                        "only 2 non-keyword arguments accepted");
+                        "only 2 non-keyword arguments accepted in np.array.");
```

Comment:
How about we match the normal python message here:
```
def array(object, dtype=None, *, copy=True, order='K', subok=False, ndmin=0):
    pass
```
```
>>> array(1, 2, 3)
TypeError: array() takes from 1 to 2 positional arguments but 3 were given
```
so
```suggestion
        PyErr_Format(PyExc_TypeError,
                     "array() takes from 1 to 2 positional arguments but %zd were given", PyTuple_GET_SIZE(args));
```

Reply:
It would be great if you could add the `*` before `copy` like I do above on this line too:

https://github.com/numpy/numpy/blob/0b9b176dcef7f14f56dbd543c89e6da6c6b7b2e4/numpy/core/_add_newdocs.py#L787